azure_functions_worker/bindings/shared_memory_data_transfer/shared_memory_constants.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+
+class SharedMemoryConstants:
+    class HeaderFlags:
+        """
+        Flags that are present in the header region of the memory maps.
+        """
+        # Indicates that the memory map has been initialized, may be in use and
+        # is not new.
+        # This represents a boolean value of True.
+        Initialized = b'\x01'
+
+    # The length of a bool which is the length of the part of the header flag
+    # specifying if the memory map is already created and used.
+    # This is to distinguish between new memory maps and ones that were
+    # previously created and may be in use already.
+    # Header flags are defined in the class SharedMemoryConstants.HeaderFlags.
+    MEM_MAP_INITIALIZED_FLAG_NUM_BYTES = 1
+
+    # The length of a long which is the length of the part of the header
+    # specifying content length in the memory map.
+    CONTENT_LENGTH_NUM_BYTES = 8
+
+    # The total length of the header
+    CONTENT_HEADER_TOTAL_BYTES = MEM_MAP_INITIALIZED_FLAG_NUM_BYTES + \
+        CONTENT_LENGTH_NUM_BYTES
+
+    # Minimum size (in number of bytes) an object must be in order for it to be
+    # transferred over shared memory.
+    # If the object is smaller than this, gRPC is used.
+    # Note: This needs to be consistent among the host and workers.
+    #       e.g. in the host, it is defined in SharedMemoryConstants.cs
+    MIN_BYTES_FOR_SHARED_MEM_TRANSFER = 1024 * 1024  # 1 MB
+
+    # Maximum size (in number of bytes) an object must be in order for it to be
+    # transferred over shared memory.
+    # This limit is imposed because initializing objects like greater than 2GB
+    # is not allowed in DotNet.
+    # Ref: https://stackoverflow.com/a/3944336/3132415
+    # Note: This needs to be consistent among the host and workers.
+    #       e.g. in the host, it is defined in SharedMemoryConstants.cs
+    MAX_BYTES_FOR_SHARED_MEM_TRANSFER = 2 * 1024 * 1024 * 1024  # 2 GB
+
+    # This is what the size of a character is in DotNet. Can be verified by
+    # doing "sizeof(char)".
+    # To keep the limits consistent, when determining if a string can be
+    # transferred over shared memory, we multiply the number of characters
+    # by this constant.
+    # Corresponding logic in the host can be found in SharedMemoryManager.cs
+    SIZE_OF_CHAR_BYTES = 2
+
+    # Directories in Unix where the memory maps can be found
+    UNIX_TEMP_DIRS = ["/dev/shm"]
```

Comment:
A temp file approach won't give the same performance. The test would pass, but we want to still use memory-based file system to get the performance we want when running a real function.
From some basic searching, it seems like you can create a memory-based file system on macOS (so we can use the same path for that `/dev/shm` and keep things consistent.) Reference [here](https://stackoverflow.com/a/2033417/3132415).

The idea would be to do this as part of some install/setup thing when run on macOS? What do you think?
Because even on a non-prod scenario when a user is running this locally, even for Linux, we would want to create `/dev/shm` or assign some space to it (if it is not present, nothing would fail hopefully and RPC would continue to be used). Should that be some custom install step or what?

Reply:
In addition to fixing this, we probably should add other local folders like /tmp